src/ucp/core/ucp_am.c

```diff
@@ -585,14 +635,32 @@ static ucs_status_t ucp_am_contig_short(uct_pending_req_t *self)
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucp_ep_t *ep       = req->send.ep;
     ucs_status_t status;
+    ucp_request_t sreq_tmp;
```

Comment:
logic around `sreq_tmp` looks weird to me, I guest this is related to setting/unsetting flags

Reply:
It's for releasing the header back to the mpool.
ucp_am_short_handle_status_from_pending can call ucp_request_complete_send that release the request with the header pointer, so I keep a temporary pointer for releasing the header.

I've done the same thing in ucp_proto_progress_am_rndv_rts, there I think it's needed, just decided to do the same here, but I can check the send operation status before calling the handle_status function and get rid of the temp pointer.

Reply:
maybe just add another parameter to `ucp_am_short_handle_status_from_pending`? it could be constant in other places and compiled out.

Reply:
This function is used by other protocols that make use of the AM API.
I'm not sure If adding another unique parameter is better than simply moving this branch before ucp_am_short_handle_status_from_pending.

Reply:
how about another `ucp_am_short_handle_status_from_pending` wrapper for this particular protocol?

My concern about `sreq_tmp`, not about where exactly the code should be placed.